driver/csiplugin/nodeserver.go

```diff
@@ -95,6 +96,17 @@ func (ns *ScaleNodeServer) NodePublishVolume(ctx context.Context, req *csi.NodeP
 		}
 	} else {
 		//Use bind mount
+		volScalePathInContainer := hostDir + volScalePath
+		args := []string{"-f", "-c", "%T", volScalePathInContainer}
+		out, err := executeCmd("stat", args)
```

Comment:
I realized that, blocking use of symlink works for static PVs and version 2 PVs (PVs with CG). But for the classic PVs (version 1), we create a symlink in primary fileset to the actual fileset data directory(this is a different symlink than the one which is deprecated and bind mount is being used as an alternative currently). So we cannot block usage of symlink. So now, to have the support for symlink and block symlinks whose target is of type other than "gpfs", I have made a change to follow the symlink to get the target path and get the type of target path and allow only when the target is of type "gpfs."
	
Also I checked if I can lock the symlink file after obtaining the file descriptor, but I see when we do openFile() for symlink file, instead of actual symlink, it tries to open the target of symlink, so I couldn't put a lock on the symlink file using fcntl. @dunnevan Please let me know if you have any suggestions on this. Thanks!

Reply:
Need to explore more on the locking part, will check more and handle it in another PR.